src/ArduinoCellular.cpp

```diff
@@ -205,24 +205,32 @@ SimStatus ArduinoCellular::getSimStatus(){
         this->debugStream->println("SIM Status: " + String(simStatus));
     }
 
-    if (modem.getSimStatus() == 0) {
+    if (simStatus == 0) {
         return SimStatus::SIM_ERROR;
-    } else if (modem.getSimStatus() == 1) {
+    } else if (simStatus == 1) {
         return SimStatus::SIM_READY;
-    } else if (modem.getSimStatus() == 2) {
+    } else if (simStatus == 2) {
         return SimStatus::SIM_LOCKED;
-    } else if (modem.getSimStatus() == 3) {
+    } else if (simStatus == 3) {
         return SimStatus::SIM_ANTITHEFT_LOCKED;
     } else {
         return SimStatus::SIM_ERROR;
     }
 }
 
 bool ArduinoCellular::unlockSIM(String pin){
-    if(this->debugStream != nullptr){
-        this->debugStream->println("Unlocking SIM...");
+    int simStatus = modem.getSimStatus();
```

Comment:
isn't this shadowing the global `simStatus` ?

Reply:
Sorry, I had an issue with visualization, everything is ok :slightly_smiling_face: